sympy/core/basic.py

```diff
@@ -232,7 +232,7 @@ def compare(self, other):
             r = Basic(*r) if isinstance(r, frozenset) else r
             if isinstance(l, Basic):
                 c = l.compare(r)
-            else:
+            elif not (isinstance(l, Tuple) and isinstance(l, Tuple)):
```

Comment:
Are you sure of this `isinstance(l, Tuple) and isinstance(l, Tuple) ` doesn't make too much sense !